scripts/provisioning.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+################################################################################
+################################################################################
+#
+# This script calls the needed DebOps plays/roles in order to provision a new cloud VM,
+# then prepares the VM to be ansible managed and runs additional Ansible Playbooks in the correct order.
+#
+# Usage:
+#     ./provisioning.sh -l <host-or-group-to-provision>
+#
+#     You can also pass ansible parameter to DebOps
+#     ./provisioning.sh -l <host-or-group-to-provision> -t role::pki --skip-tags role::owncloud --user root -vvv
+#
+################################################################################
+
+set -o nounset -o pipefail -o errexit
+
+# Check if we are in the correct root folder of the repo, and change upwards if needed
+# otherwise debops won't work
+if [[ ! -f .debops.cfg && -f ../.debops.cfg ]]; then
+    echo "Running $0 from root of repo."
+    pushd .. > /dev/null
+fi
+
+# Run the bootstrap DebOps playbook to make the VM a DebOps/Ansible managed host.
+debops --become $@ bootstrap-ldap || debops --become --user root $@ bootstrap-ldap
+
+# Execute all DebOps plays (site.yml) to install everything.
+debops $@
+
+# Run the PKI role a second time (the first is executed from the site.yml DebOps playbook)
+#   The second run is necessary for DebOps to download/request the Let's Encrypt ACME TLS certificates
+#   Somehow during the first run, only the internal CA TLS Certificates are installed, but not the ACME certificates.
```

Comment:
Well, there's no "somehow" - the `debops.pki` role needs the `nginx` server configured with `debops.nginx` role to handle the ACME `http-01` authentication request. So, on the first run, Let's Encrypt certificates cannot be acquired because `nginx` server isn't ready to help authenticate the request.

Besides, getting the Let's Encrypt certificates work in a very specific environment:
- `nginx` server is configured
- the host has at least 1 public IP address
- the DNS configuration is sound, and the host has a Fully Qualified Domain Name set in DNS
- a separate PKI realm is configured; the default `domain` PKI realm has ACME CA disabled intentionally because of the above reasons

So, adding another run of the `service/pki` playbook here might be ill-advised, I think. DebOps provides a set of internal certificates that work reasonably well, use of Let's Encrypt certificates should be a conscious decision, especially that each LE certificate is registered in a public directory. I don't think that DebOps users would like to have their experimental domains suddenly exposed to the outside world like that...

Reply:
See https://github.com/debops/examples/pull/15#discussion_r333633228